core/state/snapshot/disklayer.go

```diff
@@ -133,6 +181,50 @@ func (dl *diskLayer) AccountRLP(hash common.Hash) ([]byte, error) {
 func (dl *diskLayer) Storage(accountHash, storageHash common.Hash) ([]byte, error) {
 	dl.lock.RLock()
 	defer dl.lock.RUnlock()
+	start := time.Now()
+
+	routeid := cachemetrics.Goid()
+	hitInL3 := false
+	hitInDisk := false
+	var startGetInDisk time.Time
+	defer func() {
+		isSyncMainProcess := cachemetrics.IsSyncMainRoutineID(routeid)
+		isMinerMainProcess := cachemetrics.IsMinerMainRoutineID(routeid)
+		if isSyncMainProcess {
+			// layer 2 miss
+			syncL2StorageMissMeter.Mark(1)
+			if hitInL3 {
+				syncL3StorageHitMeter.Mark(1)
+				cachemetrics.RecordCacheDepth("CACHE_L3_STORAGE")
+				cachemetrics.RecordCacheMetrics("CACHE_L3_STORAGE", start)
+				cachemetrics.RecordTotalCosts("CACHE_L3_STORAGE", start)
+			}
+			if hitInDisk {
+				// layer 3 miss
+				syncL3StorageMissMeter.Mark(1)
+				cachemetrics.RecordCacheDepth("DISK_L4_STORAGE")
+				cachemetrics.RecordCacheMetrics("DISK_L4_STORAGE", startGetInDisk)
+				cachemetrics.RecordTotalCosts("DISK_L4_STORAGE", start)
+			}
+		}
+		if isMinerMainProcess {
+			// layer 2 miss
+			minerL2StorageMissMeter.Mark(1)
```

Comment:
move `minerL2StorageMissMeter.Mark(1)` into  `if hitInL3 {`

Reply:
no need move it , same reason  as above